### PR TITLE
Fix bug where passing existing experiment name into UpdateChaosExperiment causes error

### LINKED ISSUE #5320: Calling UpdateChaosExperiment with same name returns error
**What happened**:
When calling the `updateChaosExperiment` mutation and passing in the same experiment name that the experiment already had, an error is thrown, stating that it's a duplicate name.

**What you expected to happen**:
The mutation should work as usual with no changes to the experiment name.

**Where can this issue be corrected? (optional)**

In `chaoscenter/graphql/server/pkg/chaos_experiment/handler/handler.go`, there is logic to catch duplicate experiment names in `UpdateChaosExperiment`, but it throws an error even if the duplicate experiment name belongs to the original experiment.

**How to reproduce it (as minimally and precisely as possible)**:
Call `updateChaosExperiment` mutation with the experiment name field as the experiment's existing name.

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by 1spyral**
Fixes #5320 

## Proposed changes

Fetches the existing experiments with the declared experiment name and compares it with the experiment being edited. If they are the same, skip the duplicate name check which was previously erroring.

Changes to tests:
- Mock the fetch for the existing experiment
- Rename tests to use `success` and `failure` like other tests in file
- Add test to ensure that passing a duplicate experiment name throws an error

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: chaoscenter/graphql/server/pkg/chaos_experiment/handler/handler.go [hunk 1]
```diff
@@ -283,12 +283,30 @@ func (c *ChaosExperimentHandler) UpdateChaosExperiment(ctx context.Context, requ
 		revID = uuid.New().String()
 	)
 
-	// Check if the experiment_name exists under same project
-	err := c.validateDuplicateExperimentName(ctx, projectID, request.ExperimentName)
+	// Fetch the existing experiment
```
Comment (SarthakJain26) [inline chaoscenter/graphql/server/pkg/chaos_experiment/handler/handler.go]: This can further be optimised, by fetching the mongo document by querying it based on name. Then check if the number of documents is more than 1 then
- if request.ExperimentID == wfDetails.ExperimentID which means same document is being modified so the validation can be skipped
- if request.ExperimentID != wfDetails.ExperimentID that means another document exists with the same name so in this case it should throw an error that same name exists in the project.

 With this method you can skip using the validateDuplicateExperimentName and save an extra DB call

**Comment by 1spyral** [comment]
@SarthakJain26 thanks for the review! I've optimized the check to use one DB query. Let me know if everything looks good to you!

**Review by SarthakJain26 — APPROVED** [review]

**Review by PriteshKiri — APPROVED** [review]

**Comment by 1spyral** [comment]
Fixed outdated mocks in fuzz test

**Comment by SarthakJain26** [comment]
graphq-server build pipeline failing due to vulnerability scanning. Will fix the vulnerability in another PR.
